src/PIL/Image.py

```diff
@@ -203,48 +203,16 @@ def isImageType(t):
 # --------------------------------------------------------------------
 # Modes
 
-if sys.byteorder == "little":
-    _ENDIAN = "<"
-else:
-    _ENDIAN = ">"
-
-_MODE_CONV = {
-    # official modes
-    "1": ("|b1", None),  # Bits need to be extended to bytes
-    "L": ("|u1", None),
-    "LA": ("|u1", 2),
-    "I": (_ENDIAN + "i4", None),
-    "F": (_ENDIAN + "f4", None),
-    "P": ("|u1", None),
-    "RGB": ("|u1", 3),
-    "RGBX": ("|u1", 4),
-    "RGBA": ("|u1", 4),
-    "CMYK": ("|u1", 4),
-    "YCbCr": ("|u1", 3),
-    "LAB": ("|u1", 3),  # UNDONE - unsigned |u1i1i1
-    "HSV": ("|u1", 3),
-    # I;16 == I;16L, and I;32 == I;32L
-    "I;16": ("<u2", None),
-    "I;16B": (">u2", None),
-    "I;16L": ("<u2", None),
-    "I;16S": ("<i2", None),
-    "I;16BS": (">i2", None),
-    "I;16LS": ("<i2", None),
-    "I;32": ("<u4", None),
-    "I;32B": (">u4", None),
-    "I;32L": ("<u4", None),
-    "I;32S": ("<i4", None),
-    "I;32BS": (">i4", None),
-    "I;32LS": ("<i4", None),
-}
+_ENDIAN = "<" if sys.byteorder == "little" else ">"
```

Comment:
You know better than me for what this is used later on, but if you only need this to set the native byte order for numpy, you could also use `_ENDIAN= "="` (`=` for native) and have numpy do the rest. Of course, you could also set it directly in the typestring, e.g. `=u4`.

Reply:
While that is listed at https://numpy.org/doc/stable/reference/generated/numpy.dtype.byteorder.html, it is not listed under "typestr" at https://numpy.org/doc/stable/reference/arrays.interface.html#object.__array_interface__. So I'm skeptical that "=" would be valid.

Reply:
Good catch. So far I used them interchangeably since I never noticed something being broken, but I agree it is better to err on the side of caution and not use it.